packages/@n8n/config/src/configs/database.config.ts

```diff
@@ -75,6 +75,10 @@ class PostgresConfig {
 	@Env('DB_POSTGRESDB_POOL_SIZE')
 	poolSize: number = 2;
 
+	/** Postgres connection timeout (ms) */
+	@Env('DB_POSTGRESDB_CONNECTION_TIMEOUT')
+	connectionTimeoutMs: number = 5000;
```

Comment:
since this is the timeout for the initial connection handshake, does this really need to be this long? could we reduce this to 500ms or 1 second?